gdal/frmts/nitf/nitffile.c

```diff
@@ -2301,23 +2302,30 @@ static char** NITFGenericMetadataReadTREInternal(char **papszMD,
                         break;
                     }
                 }
-                else if (strcmp(pszType, "bitmask") == 0)
+                else if (strcmp(pszType, "UnsignedInt_BigEndian") == 0 || strcmp(pszType, "bitmask") == 0)
                 {
-                    if( nLength == 4 )
+                    if (nLength <= 8)
                     {
-                        const size_t nBufferSize = 11;
-                        unsigned int nVal;
-                        memcpy(&nVal, pachTRE + *pnTreOffset, 4);
-                        CPL_MSBPTR32(&nVal);
+                        const size_t nBufferSize = 21;
+                        unsigned long long nVal = 0;
+                        GByte byData;
+
+                        int i;
+                        for (i = 0; i < nLength; ++i)
+                        {
+                            memcpy(&byData, pachTRE + *pnTreOffset + i, 1);
+                            nVal += (unsigned long long)byData << 8*(nLength - i - 1);
+                        }
+
                         pszValue = (char*)CPLMalloc(nBufferSize);
-                        CPLsnprintf(pszValue, nBufferSize, "%u", nVal);
+                        CPLsnprintf(pszValue, nBufferSize, "%llu", nVal);
```

Comment:
to fix this warning on mingw64
```
nitffile.c:2321:25: error: unknown conversion type character ‘l’ in format [-Werror=format=]
                         CPLsnprintf(pszValue, nBufferSize, "%llu", nVal);
```
maybe try:
```suggestion
                        CPLsnprintf(pszValue, nBufferSize, CPL_FRMT_GUIB, (GUIntBig)nVal);
```